pkg/analytics/analytics.go

```diff
@@ -2,57 +2,114 @@ package analytics
 	semconv "go.opentelemetry.io/otel/semconv/v1.26.0"
 
 	"github.com/bacalhau-project/bacalhau/pkg/models"
+	"github.com/bacalhau-project/bacalhau/pkg/version"
 )
 
-type Recorder interface {
-	EmitEvent(ctx context.Context, event EventType, properties ...otellog.KeyValue)
-	EmitJobEvent(ctx context.Context, event EventType, j models.Job)
-	Stop(ctx context.Context) error
+const ProviderKey = "bacalhau-analytics"
+
+const (
+	NodeInstallationIDKey = "installation_id"
+	NodeInstanceIDKey     = "instance_id"
+	NodeIDKey             = "node_id"
+	NodeTypeKey           = "node_type"
+	NodeVersionKey        = "node_version"
+)
+
+type Config struct {
+	attributes   []attribute.KeyValue
+	otlpEndpoint string
+}
+
+type Option func(*Config)
+
+func WithEndpoint(endpoint string) Option {
+	return func(c *Config) {
+		c.otlpEndpoint = endpoint
+	}
+}
+
+func WithNodeNodeID(id string) Option {
+	return func(c *Config) {
+		c.attributes = append(c.attributes, attribute.String(NodeIDKey, id))
+	}
+}
+
+func WithNodeType(isRequester, isCompute bool) Option {
+	return func(c *Config) {
+		var typ string
+		if isRequester && isCompute {
+			typ = "hybrid"
+		} else if isRequester {
+			typ = "orchestrator"
+		} else if isCompute {
+			typ = "compute"
+		}
+		c.attributes = append(c.attributes, attribute.String(NodeTypeKey, typ))
+	}
 }
 
-var _ Recorder = (*LogRecorder)(nil)
+func WithInstallationID(id string) Option {
+	return func(c *Config) {
+		c.attributes = append(c.attributes, attribute.String(NodeInstallationIDKey, id))
+	}
+}
 
-type LogRecorder struct {
-	provider *sdklog.LoggerProvider
+func WithInstanceID(id string) Option {
+	return func(c *Config) {
+		c.attributes = append(c.attributes, attribute.String(NodeInstanceIDKey, id))
+	}
 }
 
-func New(ctx context.Context, opts ...Option) (*LogRecorder, error) {
+func WithVersion(bv *models.BuildVersionInfo) Option {
+	return func(c *Config) {
+		v, err := semver.NewVersion(bv.GitVersion)
+		if err != nil {
+			// use the version populated via the `ldflags` flag.
+			c.attributes = append(c.attributes, attribute.String(NodeVersionKey, version.GITVERSION))
+		} else {
+			c.attributes = append(c.attributes, attribute.String(NodeVersionKey, v.String()))
+		}
+	}
+}
+
+func SetupAnalyticsProvider(ctx context.Context, opts ...Option) error {
 	config := &Config{
-		otlpEndpoint: "localhost:4317", // Default endpoint
+		otlpEndpoint: "localhost:4318", // Default endpoint
 		attributes:   make([]attribute.KeyValue, 0),
 	}
 	// Apply options
 	for _, opt := range opts {
-		if err := opt(config); err != nil {
-			return nil, fmt.Errorf("failed to apply option: %w", err)
-		}
+		opt(config)
 	}
 
 	// Create the file exporter
 	// TODO before merging we'll need to disable this
 	stdoutExporter, err := stdoutlog.New()
 	if err != nil {
-		return nil, fmt.Errorf("failed to create stdout exporter: %w", err)
+		return fmt.Errorf("failed to create stdout exporter: %w", err)
 	}
```

Comment:
do you still need this expoerter?

Reply:
No, it's been removed in the latest commit.